File: nmaas-portal/src/app/model/domain.ts
import {DomainDcnDetails} from './domaindcndetails';
import {DomainTechDetails} from './domaintechdetails';
import {DomainApplicationStatePerDomain} from './domainapplicationstateperdomain';

export class Domain {
  public id: number = undefined;
  public name: string = undefined;
  public codename: string = undefined;
  public active: boolean = undefined;
  public domainDcnDetails: DomainDcnDetails = new DomainDcnDetails();
  public domainTechDetails: DomainTechDetails = new DomainTechDetails();
  public applicationStatePerDomain: DomainApplicationStatePerDomain[] = [];
}

File: nmaas-platform/src/main/java/net/geant/nmaas/portal/api/domain/Domain.java
package net.geant.nmaas.portal.api.domain;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Domain {
	Long id;

	String name;
	String codename;
	boolean active;
	boolean dcnConfigured;
	String kubernetesNamespace;
	String kubernetesStorageClass;
}
